whois-api/src/test/java/net/ripe/db/whois/api/httpserver/ClientCertificateServiceTestIntegration.java

```diff
@@ -3,40 +3,44 @@
 import jakarta.ws.rs.ProcessingException;
 import jakarta.ws.rs.core.MediaType;
 import net.ripe.db.whois.api.SecureRestTest;
+import net.ripe.db.whois.common.aspects.RetryFor;
 import org.junit.jupiter.api.Tag;
 import org.junit.jupiter.api.Test;
 import org.springframework.test.annotation.DirtiesContext;
 
+import java.io.IOException;
+
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.containsString;
-import static org.junit.jupiter.api.Assertions.fail;
+import static org.junit.jupiter.api.Assertions.assertThrows;
 
 @Tag("IntegrationTest")
 @DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD)
 public class ClientCertificateServiceTestIntegration extends AbstractClientCertificateIntegrationTest {
 
     @Test
+    @RetryFor(value = IOException.class, attempts = 10, intervalMs = 10000)
     public void client_certificate() {
-        final String response = SecureRestTest.target(getClientSSLContext(), getClientCertificatePort(), "whois/client")
-            .request()
-            .get(String.class);
-
+        final String response = getClientCertificateWithCertificate();
         assertThat(response, containsString("Found 1 certificate(s)."));
         assertThat(response, containsString("-----BEGIN CERTIFICATE-----"));
         assertThat(response, containsString("-----END CERTIFICATE-----"));
+
+        final ProcessingException processingException = assertThrows(ProcessingException.class,
```

Comment:
Why are you expecting a `ProcessingException` ? The test should succeed

Reply:
Should the test retry on `ProcessingException` ? For example
```
@RetryFor(value = {IOException.class, ProcessingException.class}, attempts = 10, intervalMs = 10000)
```

Reply:
This is no fixing 100% the test. It is a draft, please do not review it